dali/python/nvidia/dali/pipeline.py

```diff
@@ -75,17 +84,30 @@ def __init__(self, batch_size = -1, num_threads = -1, device_id = -1, seed = -1,
         self._device_id = device_id
         self._seed = seed
         self._exec_pipelined = exec_pipelined
-        self._prefetch_queue_depth = prefetch_queue_depth
         self._built = False
         self._first_iter = True
         self._last_iter = False
         self._batches_to_consume = 0
+        self._cpu_batches_to_consume = 0
+        self._gpu_batches_to_consume = 0
         self._prepared = False
         self._names_and_devices = None
         self._exec_async = exec_async
         self._bytes_per_sample = bytes_per_sample
         self._set_affinity = set_affinity
         self._max_streams = max_streams
+        if type(prefetch_queue_depth) is dict:
+            self._exec_separated = True
+            self._cpu_queue_size = prefetch_queue_depth["cpu_size"]
+            self._gpu_queue_size = prefetch_queue_depth["gpu_size"]
+            self._prefetch_queue_depth = self._cpu_queue_size  # dummy value, that will be ignored
```

Comment:
Why do you assign it then?

Reply:
Because I still have to pass something to c++ Pipeline in that place. And I than change it by `SetExecutionTypes` and `SetQueueSizes`